interp/run.go

```diff
@@ -3133,19 +3133,27 @@ func convertConstantValue(n *node) {
 // Write to a channel.
 func send(n *node) {
 	next := getExec(n.tnext)
-	value0 := genValue(n.child[0]) // channel
-	convertLiteralValue(n.child[1], n.child[0].typ.val.TypeOf())
-	value1 := genValue(n.child[1]) // value to send
+	c0, c1 := n.child[0], n.child[1]
+	value0 := genValue(c0) // Send channel.
+	convertLiteralValue(c1, c0.typ.val.TypeOf())
+
+	var value1 func(*frame) reflect.Value // Value to send.
+	switch {
+	case isInterfaceBin(c0.typ.val):
+		value1 = genInterfaceWrapper(c1, c0.typ.val.rtype)
+	default:
+		value1 = genValue(c1)
+	}
 
 	if n.interp.cancelChan {
```

Comment:
we could 1) invert that test, and 2) return early, for the least amount of indented code.